### PR TITLE
First component created

### PR DESCRIPTION
**Pull request by vibhu1604**
First component added

## Summary by CodeRabbit

- **New Features**
  - Introduced a new alert component that allows customizable severity, title, and message.
  - Added interactive buttons to the new alert component.
  
- **Integration**
  - Integrated the new alert component into the Monitors page.

### CODE REVIEW

**Review by danielcj2 — CHANGES_REQUESTED** [review]
Seems good, just a couple of changes and it should be ready to go!

It looks like you're missing documentation. I would recommend looking over some of the other components and familiarize yourself with `prop-types` and `jsdocs`. They are fairly easy to use and make the code easier to understand and maintain.

**Comment by ajhollid** [comment]
Good call on the docs.

@vibhu1604 it may seem trivial, but if we add the documentation and proptypes as we go we have a much easier time than if we have to go through 50 different components and add the docs all at once.

```
import { Button as MuiButton } from "@mui/material";
import PropTypes from "prop-types";

const levelConfig = {
  primary: {
    variant: "contained",
    color: "primary",
  },
  secondary: {
    variant: "outlined",
    color: "secondary",
  },
  tertiary: {
    variant: "text",
    color: "tertiary",
  },
  error: {
    variant: "contained",
    color: "error",
  },
  imagePrimary: {
    color: "primary",
    variant: "text",
  },
  imageSecondary: {
    color: "secondary",
    variant: "text",
  },
  imageTertiary: {
    color: "tertiary",
    variant: "text",
  },
};

/**
 * @component
 * @param {Object} props
 * @param {'primary' | 'secondary' | 'tertiary' | 'error' | 'imagePrimary' | 'imageSecondary' | 'imageTertiary'} props.level - The level of the button
 * @param {string} props.type - The type of the button
 * @param {string} props.label - The label of the button
 * @param {React.ReactNode} props.img - Image for button
 * @param {boolean} [props.disabled] - Whether the button is disabled
 * @param {function} props.onClick - Function to run when the button is clicked
 * @param {Object} props.sx - Styles for the button
 * @returns {JSX.Element}
 * @example
 * // Render an error button
 * <Button type="submit" level="error" label="Error" disabled sx={{marginTop: "1rem"}}/>
 */

const Button = ({ id, type, level, label, disabled, img, onClick, sx }) => {
  const { variant, color } = levelConfig[level];
  return (
    <MuiButton
      id={id}
      type={type}
      variant={variant}
      color={color}
      disabled={disabled}
      onClick={onClick}
      sx={{
        textTransform: "none",
        "&:focus": {
          outline: "none",
        },
        ...sx,
      }}
    >
      {img && img}
      {label}
    </MuiButton>
  );
};

Button.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
  level: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  img: PropTypes.node,
  onClick: PropTypes.func,
  sx: PropTypes.object,
  disabled: PropTypes.bool,
};

export default Button;

```

Here's an example of a `Button` that has documentation and proptypes

**Comment by vibhu1604** [comment]
Ok got it. I will take a look on it. thanks @ajhollid & @danielcj2 .

**Comment by Skorpios604** [comment]
Ya looks good to me other than the things @ajhollid and @danielcj2 mentioned. Thanks!

File: Client/src/Components/VibhutiComponent/index.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <Stack spacing={0} direction="row">
+        <Button color="secondary" variant="text">
+          Dismiss
+        </Button>
+        <Button color="primary" variant="text">
+          View Changes
+        </Button>
+      </Stack>
+    </Alert>
+  );
```
[the remark this answers is not in the document]
Comment (vibhu1604) [reply Client/src/Components/VibhutiComponent/index.jsx:27]: Do I need to write the code to make them functional or are you advising me to just add onClick event handler ?

File: Client/src/Components/VibhutiComponent/index.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+FirstComponent.propTypes = {
+  severity: PropTypes.string,
+  title: PropTypes.string,
+  message: PropTypes.string,
+};
```
[the remark this answers is not in the document]
Comment (vibhu1604) [reply Client/src/Components/VibhutiComponent/index.jsx:33]: noted

**Comment by ajhollid** [comment]
I think we can close this, this exercise is complete :+1:

**Pull request closed without merging**